torch/fx/experimental/symbolic_shapes.py

```diff
@@ -547,20 +567,37 @@ def create_symbol(self, val: int, *, simplify: bool = True) -> "sympy.Expr":
         # TODO: Create a guard whenever this happens
         # TODO: Do this duck sizing lazily later
 
-        # This implements duck-shaping: input sizes that match are assigned
-        # the same symint
+        # Create a duck sized int if necessary
         if val not in self.val_to_var:
             sympy_expr = sympy.Symbol(f"s{len(self.var_to_val)}", positive=True, integer=True)
             self.var_to_val[sympy_expr] = sympy.Integer(val)
             self.val_to_var[val] = sympy_expr
             self.definitely_not_01.add(sympy_expr)
 
-        self.replacements[symbol] = self.val_to_var[val]
+        # This implements duck-shaping: input sizes that match are assigned
+        # the same symint
+        self.replacements[symbol] = self.duck_int(val)
 
         # Return the *symbol*; you're expected to apply the replacement to get
         # the simplified variable
         return symbol
 
+    # Given a concrete integer value, return the duck sized symbol associated
+    # with it; e.g., suppose we already have a tensor of size 3 in scope,
+    # which was assigned s3, then shape_env.duck_int(3) we will get back s3.
```

Comment:
hyper-ultra-nit: change one of the digits in the ex so folks dont think s3 is 3, s2 is 2, etc.